src/common/AppNavbarMobile.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div class="app-navbar-mobile">
+    <app-logo />
+    <button
+      class="app-navbar-mobile__burger-btn"
+      @click="isExtShown = !isExtShown"
+    >
+      <transition name="fade" mode="out-in">
+        <icon
+          class="app-navbar-mobile__burger-icon"
+          :key="String(isExtShown)"
+          :name="!isExtShown ? $icons.menuAlt1 : $icons.x"
+        />
+      </transition>
+    </button>
+    <transition name="fade" appear>
+      <div
+        v-if="isExtShown"
+        ref="backdropElement"
+        class="app-navbar-mobile__ext-wrp"
+        @click="onBackdropClick"
+      >
+        <transition name="slide" appear>
+          <div class="app-navbar-mobile__ext">
+            <nav class="app-navbar-mobile__nav">
+              <app-button
+                v-for="(link, idx) in links"
+                :key="idx"
+                :text="link.text"
+                :href="link.href"
+                :route="link.route"
+                scheme="link"
+                color="none"
+                class="app-navbar-mobile__link"
+                @click="closeExt"
+              />
+            </nav>
+            <app-button
+              class="app-navbar-mobile__connect-wallet-btn"
+              color="secondary"
+              :text="$t('app-navbar.connect-wallet-btn')"
+            />
+          </div>
+        </transition>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useContext, useNavLinks } from '@/composables'
+import AppButton from './AppButton.vue'
+import AppLogo from './AppLogo.vue'
+import Icon from './Icon.vue'
+import { ref } from 'vue'
+
+const { $t } = useContext()
+const { links } = useNavLinks()
+
+const isExtShown = ref(false)
+const backdropElement = ref<HTMLDivElement | null>(null)
+
+const closeExt = () => {
+  isExtShown.value = false
+}
+
+const onBackdropClick = (event: Event) => {
+  if (!backdropElement.value) throw new Error('backdropElement unavailable')
+  if ((event.target as HTMLDivElement) === backdropElement.value) closeExt()
```

Comment:
- no need to throw
- no need to check whether the target is our element, you listen for click on it

Reply:
done